contributing/development/core_and_modules/custom_audiostreams.rst

```diff
@@ -48,69 +48,74 @@ Therefore, playback state must be self-contained in AudioStreamPlayback.
 .. code-block:: cpp
 
     /* audiostream_mytone.h */
-
-    #include "core/reference.h"
-    #include "core/resource.h"
+    
+    #ifndef AUDIOSTREAM_MYTONE_H
+    #define AUDIOSTREAM_MYTONE_H
+    
     #include "servers/audio/audio_stream.h"
-
+    
     class AudioStreamMyTone : public AudioStream {
-        GDCLASS(AudioStreamMyTone, AudioStream)
-
+    	GDCLASS(AudioStreamMyTone, AudioStream);
+    
     private:
-        friend class AudioStreamPlaybackMyTone;
-        uint64_t pos;
-        int mix_rate;
-        bool stereo;
-        int hz;
-
+    	friend class AudioStreamPlaybackMyTone;
+    	uint64_t pos;
+    	int mix_rate;
+    	bool stereo;
+    	int hz;
+    
     public:
-        void reset();
-        void set_position(uint64_t pos);
-        virtual Ref<AudioStreamPlayback> instance_playback();
-        virtual String get_stream_name() const;
-        void gen_tone(int16_t *pcm_buf, int size);
-        virtual float get_length() const { return 0; } // if supported, otherwise return 0
-        AudioStreamMyTone();
-
+    	void reset();
+    	void set_position(uint64_t pos);
+    	virtual Ref<AudioStreamPlayback> instantiate_playback();
+    	virtual String get_stream_name() const;
+    	void gen_tone(int16_t *pcm_buf, int size);
+    	
+    	virtual double get_length() const { return 0; } // if supported, otherwise return 0
+    	AudioStreamMyTone();
+    
     protected:
-        static void _bind_methods();
+    	static void _bind_methods();
     };
+    
+    #endif
 
 .. code-block:: cpp
 
     /* audiostream_mytone.cpp */
-
+    
+    #include "audiostreamplayback_mytone.h"
     #include "audiostream_mytone.h"
-
+    
     AudioStreamMyTone::AudioStreamMyTone()
-            : mix_rate(44100), stereo(false), hz(639) {
+    		: mix_rate(44100), stereo(false), hz(639) {
     }
-
-    Ref<AudioStreamPlayback> AudioStreamMyTone::instance_playback() {
-        Ref<AudioStreamPlaybackMyTone> talking_tree;
-        talking_tree.instantiate();
-        talking_tree->base = Ref<AudioStreamMyTone>(this);
-        return talking_tree;
+    
+    Ref<AudioStreamPlayback> AudioStreamMyTone::instantiate_playback() {
+    	Ref<AudioStreamPlaybackMyTone> playback;
+    	playback.instantiate();
+    	playback->base = Ref<AudioStreamMyTone>(this);
+    	return playback;
     }
-
+    
     String AudioStreamMyTone::get_stream_name() const {
-        return "MyTone";
+    	return "Adlib";
```

Comment:
Why renaming it here?

Reply:
My mistake, It should be `return "MyTone";`.